sender.go

```diff
@@ -64,29 +65,57 @@ type (
 		tab.Carrier
 		toMsg() (*amqp.Message, error)
 	}
-)
 
-// newSender creates a new Service Bus message sender given an AMQP client and entity path
-func (h *Hub) newSender(ctx context.Context) (*sender, error) {
-	span, ctx := h.startSpanFromContext(ctx, "eh.sender.newSender")
-	defer span.End()
+	// amqpSender is the bare minimum we need from an AMQP based sender.
+	// (used for testing)
+	amqpSender interface {
+		Send(ctx context.Context, msg *amqp.Message) error
+		Close(ctx context.Context) error
+	}
 
-	s := &sender{
-		hub:         h,
-		partitionID: h.senderPartitionID,
+	// getAmqpSender should return a live sender (exactly mimics the `amqpSender()` function below)
+	// (used for testing)
+	getAmqpSender func() amqpSender
+
+	senderRetryOptions struct {
+		recoveryBackoff *backoff.Backoff
+		maxRetries      int
+	}
+)
+
+func newSenderRetryOptions() *senderRetryOptions {
+	return &senderRetryOptions{
 		recoveryBackoff: &backoff.Backoff{
 			Min:    10 * time.Millisecond,
 			Max:    4 * time.Second,
 			Jitter: true,
 		},
-		cond: sync.NewCond(&sync.Mutex{}),
+		maxRetries: 5,
```

Comment:
I don't think we can do this in track 1 (at least for now) as it's a breaking change behavior-wise.

Reply:
You're absolutely right. I'll default this to infinite (and also make sure we respect infinite!)

Reply:
Done. Infinite is now the default.